Multisig/UI/ClaimToken/TokenDistributionViewController.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  TokenDistributionViewController.swift
+//  Multisig
+//
+//  Created by Mouaz on 9/5/22.
+//  Copyright © 2022 Gnosis Ltd. All rights reserved.
+//
+
+import UIKit
+
+class TokenDistributionViewController: UIViewController {
+
+    @IBOutlet private weak var titleLabel: UILabel!
+    @IBOutlet private weak var descriptionLabel: UILabel!
+    @IBOutlet private weak var nextButton: UIButton!
+    @IBOutlet weak var distributionView: BorderedCheveronButton!
+
+    private var onNext: (() -> ())?
+    private var stepNumber: Int = 1
+    private var maxSteps: Int = 3
+
+    private var stepLabel: UILabel!
+
+    convenience init(stepNumber: Int = 1, maxSteps: Int = 4, onNext: @escaping () -> ()) {
+        self.init(namedClass: TokenDistributionViewController.self)
+        self.stepNumber = stepNumber
+        self.maxSteps = maxSteps
+        self.onNext = onNext
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        distributionView.set("Distribution details") {
+            //TODO: Show distribution details
```

Comment:
probably should call a callback so that flow can open the details

Reply:
presenting the details will be done from this point when the screen is ready, no need for callback, no?